demo/style_configurator.c

```diff
@@ -0,0 +1,815 @@
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+
+	style_rgb(ctx, "Background:", &win.background);
+
+	style_item_color(ctx, "Fixed Background:", &win.fixed_background);
+
+	style_rgb(ctx, "Border:", &win.border_color);
+	style_rgb(ctx, "Popup Border:", &win.popup_border_color);
+	style_rgb(ctx, "Combo Border:", &win.combo_border_color);
+	style_rgb(ctx, "Contextual Border:", &win.contextual_border_color);
+	style_rgb(ctx, "Menu Border:", &win.menu_border_color);
+	style_rgb(ctx, "Group Border:", &win.group_border_color);
+	style_rgb(ctx, "Tooltip Border:", &win.tooltip_border_color);
+
+	style_item_color(ctx, "Scaler:", &win.scaler);
+
+	style_vec2(ctx, "Spacing:", &win.spacing);
+	style_vec2(ctx, "Scrollbar Size:", &win.scrollbar_size);
+	style_vec2(ctx, "Min Size:", &win.min_size);
+	style_vec2(ctx, "Padding:", &win.padding);
+	style_vec2(ctx, "Group Padding:", &win.group_padding);
+	style_vec2(ctx, "Popup Padding:", &win.popup_padding);
+	style_vec2(ctx, "Combo Padding:", &win.combo_padding);
+	style_vec2(ctx, "Contextual Padding:", &win.contextual_padding);
+	style_vec2(ctx, "Menu Padding:", &win.menu_padding);
+	style_vec2(ctx, "Tooltip Padding:", &win.tooltip_padding);
+
+	nk_property_float(ctx, "#Rounding:", -100.0f, &win.rounding, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Combo Border:", -100.0f, &win.combo_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Contextual Border:", -100.0f, &win.contextual_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Menu Border:", -100.0f, &win.menu_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Group Border:", -100.0f, &win.group_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Tooltip Border:", -100.0f, &win.tooltip_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Popup Border:", -100.0f, &win.popup_border, 100.0f, 1,0.5f);
+	nk_property_float(ctx, "#Border:", -100.0f, &win.border, 100.0f, 1,0.5f);
+
+	nk_layout_row_dynamic(ctx, 30, 1);
+	nk_property_float(ctx, "#Min Row Height Padding:", -100.0f, &win.min_row_height_padding, 100.0f, 1,0.5f);
+
+	if (nk_tree_push(ctx, NK_TREE_TAB, "Window Header", NK_MINIMIZED)) {
+		style_window_header(ctx, &win.header);
+		nk_tree_pop(ctx);
+	}
+
+	*out_style = win;
+}
+
+static int
+style_configurator(struct nk_context *ctx)
+{
+	/* window flags */
+	int border = nk_true;
+	int resize = nk_true;
+	int movable = nk_true;
+	int no_scrollbar = nk_false;
+	int scale_left = nk_false;
+	nk_flags window_flags = 0;
+	int minimizable = nk_true;
+
+	/* window flags */
+	window_flags = 0;
+	if (border) window_flags |= NK_WINDOW_BORDER;
+	if (resize) window_flags |= NK_WINDOW_SCALABLE;
+	if (movable) window_flags |= NK_WINDOW_MOVABLE;
+	if (no_scrollbar) window_flags |= NK_WINDOW_NO_SCROLLBAR;
+	if (scale_left) window_flags |= NK_WINDOW_SCALE_LEFT;
+	if (minimizable) window_flags |= NK_WINDOW_MINIMIZABLE;
+
+	struct nk_style *style = &ctx->style;
+
+	// TODO better way?
+	static int initialized = nk_false;
+	static struct nk_color color_table[NK_COLOR_COUNT];
```

Comment:
An idea could be to just let the user pass in the color table.
```c
static int
style_configurator(struct nk_context *ctx, struct nk_color color_table[NK_COLOR_COUNT])
```

Reply:
Done.